bokeh/models/renderers.py

```diff
@@ -242,6 +242,18 @@ def _check_malformed_graph_source(self):
     rendered as the graph edges.
     """)
 
+    selection_mode = Enum("default", "linked", help="""
+    With the "default" selection mode, node glyphs will be selected in the same
+    way as normal glyphs. With "linked" mode, selecting a node will also select
+    the attached edges.
+    """)
+
+    inspection_mode = Enum("default", "linked", help="""
+    With the "default" inspection mode, node glyphs will be inspected in the same
+    way as normal glyphs. With "linked" mode, inspecting a node will also inspect
+    the attached edges.
+    """)
```

Comment:
I think a way to encapsulate modes would be useful. An example might be to select other nodes with a specified attribute that the selected node has. Say I have a graph representing relationships among individuals. Perhaps when I select a given node I want to see all the other nodes (i.e. individuals) that have the same type of favorite animal or all the individuals that live in the same state.